include/deal.II/lac/vector_operations_internal.h

```diff
@@ -764,9 +764,8 @@ namespace internal
     template <typename Number, typename Number2>
     struct Dot
     {
-      static const bool vectorizes =
-        std::is_same<Number, Number2>::value &&
-        (VectorizedArray<Number>::n_array_elements > 1);
+      static const bool vectorizes = std::is_same<Number, Number2>::value &&
```

Comment:
```suggestion
      static constexpr bool vectorizes = std::is_same<Number, Number2>::value &&
```
I am not 100% sure here if this requires an explicit instantiation on some compilers, similarly to what we do for `VectorizedArray::n_array_elements` - but it is worth a try.